Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeSystemComponent.cpp

```diff
@@ -64,7 +65,13 @@ namespace AzToolsFramework
 
     void FocusModeSystemComponent::SetFocusRoot(AZ::EntityId entityId)
     {
+        if (m_focusRoot == entityId)
```

Comment:
here I presume I will change to something to the effect of:

```
if (m_focusRoot == entityId)
{
    // deactivate focus mode for this viewport
    return;
}
else
{
    // deactivate focus mode for this viewport
}
```

Reply:
So, if m_focusRoot == entityId we just return because the focus root is not changing.

The only times we should enable or disable focus mode on the tracking side should be when we switch from invalid entity id to a valid one and viceversa. So something like this:

```
if (!m_focusRoot.IsValid() && entityId.IsValid())
{
    // activate focus mode for this entity context
}
else if (m_focusRoot.IsValid() && !entityId.IsValid())
{
    // deactivate focus mode for this viewport
}
```